BSPConvert.Lib/Source/EntityConverter.cs

```diff
@@ -209,6 +212,77 @@ private void ConvertFuncStatic(Entity funcStatic)
 			funcStatic.ClassName = "func_brush";
 		}
 
+		private void ConvertFuncPlat(Entity entity)
+		{
+			const int q3LipMod = 2; // Quake adds 2 units to lip for some reason
+			var moveDistance = 0f;
+			var brushThickness = GetBrushThickness(entity);
+
+			if (float.TryParse(entity["height"], out var height))
+				moveDistance = height + brushThickness;
+			else if (float.TryParse(entity["lip"], out var lip))
+				moveDistance = -(lip - q3LipMod - (brushThickness * 2));
+
+			if (string.IsNullOrEmpty(entity.Name))
+			{
+				entity.Name = $"plat{entity.ModelNumber}";
+				CreatePlatTrigger(entity);
+			}
+			entity.ClassName = "func_door";
+			entity["lip"] = moveDistance.ToString();
+			entity["movedir"] = "-90 0 0";
+			entity["spawnpos"] = "1";
+			entity["spawnflags"] = "0";
+			entity["wait"] = "-1";
+		}
+
+		private void CreatePlatTrigger(Entity entity)
+		{
+			var trigger = new Entity
+			{
+				ClassName = "trigger_multiple",
+				Model = entity.Model,
+				Spawnflags = 1,
+				Origin = new Vector3(entity.Origin.X, entity.Origin.Y, entity.Origin.Z + 2)
+			};
+			trigger["parentname"] = entity.Name;
+			sourceEntities.Add(trigger);
+
+			AddPlatTriggerConnections(entity, trigger);
+		}
+
+		private void AddPlatTriggerConnections(Entity plat, Entity trigger)
+		{
+			var connection = new Entity.EntityConnection()
+			{
+				name = "onStartTouch",
+				target = plat.Name,
+				action = "close",
+				param = null,
+				delay = 0,
+				fireOnce = -1
+			};
+			trigger.connections.Add(connection);
+
+			var connection2 = new Entity.EntityConnection()
+			{
+				name = "onFullyClosed",
+				target = plat.Name,
+				action = "open",
+				param = null,
+				delay = 3, // placeholder value TODO: replicate actual func_plat behaviour
+				fireOnce = -1
+			};
+			plat.connections.Add(connection2);
+		}
+
+		private float GetBrushThickness(Entity entity)
+		{
+			var model = q3Models[entity.ModelNumber];
```

Comment:
I think this can be simplified further to just `var model = entity.Model`

Reply:
entity.Model is just the model number in string form, only q3Models have model data as far as I can tell.

Reply:
![image](https://github.com/user-attachments/assets/b512c682-70e3-44eb-8f8f-94e703feb1d1)